Lib/concurrent/futures/thread.py

```diff
@@ -215,10 +215,20 @@ def _initializer_failed(self):
                 if work_item is not None:
                     work_item.future.set_exception(BrokenThreadPool(self._broken))
 
-    def shutdown(self, wait=True):
+    def shutdown(self, wait=True, cancel_futures=False):
         with self._shutdown_lock:
             self._shutdown = True
             self._work_queue.put(None)
+            if cancel_futures:
+                while True:
```

Comment:
Could you add a comment explaining the approach?

Reply:
Sure, I can add a comment here. The reason I didn't before is because very similar logic is used directly above in `_initializer_failed()`, it's effectively the same process of draining the queue with a few changes. But, since this is a public method, I think it makes sense to include a brief description here as well.